addons/account/models/account_analytic_account.py

```diff
@@ -24,18 +24,20 @@ def _compute_invoice_count(self):
             ('move_id.move_type', 'in', sale_types),
         ])
         query.add_where(
-            'account_move_line.analytic_distribution ?| %s',
+            'ARRAY[%s] && array_remove(regexp_split_to_array(jsonb_path_query_array(account_move_line.analytic_distribution, \'$.keyvalue()."key"\')::text, \'\\D+\'), \'\')',
             [[str(account_id) for account_id in self.ids]],
         )
 
         query_string, query_param = query.select(
-            'jsonb_object_keys(account_move_line.analytic_distribution) as account_id',
-            'COUNT(DISTINCT(account_move_line.move_id)) as move_count',
+            'string_to_array(array_to_string(array_agg(array_to_string(array_remove(regexp_split_to_array(jsonb_path_query_array(account_move_line.analytic_distribution, \'$.keyvalue()."key"\')::text, \'\\D+\'), \'\'), \',\')), \',\'), \',\') as account_ids',
         )
-        query_string = f"{query_string} GROUP BY jsonb_object_keys(account_move_line.analytic_distribution)"
+        query_string = f"{query_string} GROUP BY account_move_line.move_id"
 
         self._cr.execute(query_string, query_param)
-        data = {int(record.get('account_id')): record.get('move_count') for record in self._cr.dictfetchall()}
+        data = {}
+        for record in self._cr.dictfetchall():
+            for account_id in list(dict.fromkeys(record['account_ids'])):
+                data[int(account_id)] = 1 + (data.get(int(account_id)) or 0)
```

Comment:
Perhaps we could improve a little bit the query by moving the count in sql, so that we don't need too much data manipulation
```diff
diff --git a/addons/account/models/account_analytic_account.py b/addons/account/models/account_analytic_account.py
index 751b303c0c5c..bd33ad3c33a4 100644
--- a/addons/account/models/account_analytic_account.py
+++ b/addons/account/models/account_analytic_account.py
@@ -50,20 +50,21 @@ class AccountAnalyticAccount(models.Model):
             ('move_id.move_type', 'in', purchase_types),
         ])
         query.add_where(
-            'ARRAY[%s] && array_remove(regexp_split_to_array(jsonb_path_query_array(account_move_line.analytic_distribution, \'$.keyvalue()."key"\')::text, \'\\D+\'), \'\')',
+            '%s && regexp_split_to_array(jsonb_path_query_array(account_move_line.analytic_distribution, \'$.keyvalue()."key"\')::text, \'\\D+\')',
             [[str(account_id) for account_id in self.ids]],
         )
 
         query_string, query_param = query.select(
-            'string_to_array(array_to_string(array_agg(array_to_string(array_remove(regexp_split_to_array(jsonb_path_query_array(analytic_distribution, \'$.keyvalue()."key"\')::text, \'\\D+\'), \'\'), \',\')), \',\'), \',\') as account_ids',
+            '(regexp_matches(jsonb_agg(jsonb_path_query_array(analytic_distribution, \'$.keyvalue()."key"\'))::text, \'\\d+\', \'g\'))[1]::int as account_id'
         )
-        query_string = f"{query_string} GROUP BY account_move_line.move_id"
+        query_string = f"""
+            SELECT account_id, count(account_id) FROM
+            ({query_string}) distribution
+            GROUP BY account_id
+        """
 
         self._cr.execute(query_string, query_param)
-        data = {}
-        for record in self._cr.dictfetchall():
-            for account_id in list(dict.fromkeys(record['account_ids'])):
-                data[int(account_id)] = 1 + (data.get(int(account_id)) or 0)
+        data = {res['account_id']: res['count'] for res in self._cr.dictfetchall()}
```